src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -38,11 +41,17 @@
     private $attributesCache = array();
     private $cache = array();
 
-    public function __construct(ClassMetadataFactoryInterface $classMetadataFactory = null, NameConverterInterface $nameConverter = null, PropertyTypeExtractorInterface $propertyTypeExtractor = null)
+    /**
+     * @var ClassDiscriminatorResolverInterface|null
+     */
+    protected $classDiscriminatorResolver;
+
+    public function __construct(ClassMetadataFactoryInterface $classMetadataFactory = null, NameConverterInterface $nameConverter = null, PropertyTypeExtractorInterface $propertyTypeExtractor = null, ClassDiscriminatorResolverInterface $classDiscriminatorResolver = null)
     {
         parent::__construct($classMetadataFactory, $nameConverter);
 
         $this->propertyTypeExtractor = $propertyTypeExtractor;
+        $this->classDiscriminatorResolver = $classDiscriminatorResolver ?: (null !== $classMetadataFactory ? new ClassDiscriminatorFromClassMetadata($classMetadataFactory) : null);
```

Comment:
We try to avoid imbricating ternaries (for readability). Can you switch to a `if` construct?